Singleton/version_1/Singleton.hpp

```diff
@@ -0,0 +1,31 @@
+# ifndef SINGLETON_HPP
+# define SINGLETON_HPP
+
+# include <iostream>
+# include <mutex>
+# include <thread>
+
+
+class Singleton
+{
+	public:
+		// void				setName(std::string const&);
+		std::string			getName() const;
+		static Singleton&	getInstacne(std::string const&);
+	
+	private:
+		Singleton(std::string const&);
+
+	public:
+		Singleton(Singleton const&) = delete;
+		Singleton(Singleton&&) = delete;
+		Singleton& operator=(Singleton const&) = delete;
+		Singleton& operator=(Singleton&&) = delete;
+		~Singleton();
+
+	private:
+		std::string _name;
```

Comment:
Just for your info:

In C++ it is accepted to have members with `m_` prefix.
`_` prefix is more usable in languages like C#.

Reply:
Thanks for this too. I'll follow for this style, I like it